src/lib/salad/rtree.h

```diff
@@ -32,6 +32,7 @@
  */
 #include <stddef.h>
 #include <stdbool.h>
+#include "box/index.h"
```

Comment:
That's not right. `box` is another layer, so you shouldn't use it at all, including `index`. It should be done with a callback.